crates/rpc/engine/exchange_transition_config.rs

```diff
@@ -0,0 +1,83 @@
+use ethereum_rust_core::{serde_utils, H256};
+use ethereum_rust_storage::Store;
+use serde::{Deserialize, Serialize};
+use serde_json::Value;
+use tracing::{info, warn};
+
+use crate::utils::RpcErr;
+
+#[derive(Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct ExchangeTransitionConfigPayload {
+    #[serde(with = "serde_utils::u128::hex_str")]
+    terminal_total_difficulty: u128,
+    terminal_block_hash: H256,
+    #[serde(with = "serde_utils::u64::hex_str")]
+    terminal_block_number: u64,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct ExchangeTransitionConfigV1Req {
+    payload: ExchangeTransitionConfigPayload,
+}
+
+impl std::fmt::Display for ExchangeTransitionConfigV1Req {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(
+            f,
+            "ExchangeTransitionConfigV1Req {{ terminal_total_difficulty: {}, terminal_block_hash: {:?}, terminal_block_number: {} }}",
+            self.payload.terminal_total_difficulty,
+            self.payload.terminal_block_hash,
+            self.payload.terminal_block_number
+        )
+    }
+}
+
+impl ExchangeTransitionConfigV1Req {
+    pub fn parse(params: &Option<Vec<Value>>) -> Option<ExchangeTransitionConfigV1Req> {
```

Comment:
We have the `RpcHandler` trait to handle Rpc requests:
```Rust
trait RpcHandler: Sized {
    fn parse(params: &Option<Vec<Value>>) -> Option<Self>;

    fn call(req: &RpcRequest, storage: Store) -> Result<Value, RpcErr> {
        let request = Self::parse(&req.params).ok_or(RpcErr::BadParams)?;
        request.handle(storage)
    }

    fn handle(&self, storage: Store) -> Result<Value, RpcErr>;
}
```

You can look at other endpoints as an example

Reply:
Updated!